tests/web/test_check_clickhouse.py

```diff
@@ -0,0 +1,46 @@
+from typing import Sequence
+from unittest import mock
+
+from snuba.datasets.dataset import Dataset
+from snuba.datasets.entities import EntityKey
+from snuba.datasets.events import EventsDataset
+from snuba.web.views import check_clickhouse
+
+
+class BadStorage(mock.MagicMock):
+    def get_cluster(self) -> None:
+        raise Exception("No cluster")
+
+
+class BadEntity(mock.MagicMock):
+    def get_all_storages(self) -> Sequence[BadStorage]:
+        return [BadStorage()]
+
+
+class ExperimentalDataset(Dataset):
+    @classmethod
+    def is_experimental(cls) -> bool:
+        return True
+
+    def get_default_entity(self) -> BadEntity:
+        return BadEntity()
+
+    def get_all_entities(self) -> Sequence[BadEntity]:
+        return [BadEntity()]
+
+
+def fake_get_dataset(name: str) -> Dataset:
+    return {
+        "events": EventsDataset(),
+        "experimental": ExperimentalDataset(default_entity=EntityKey.PROFILES),
+    }[name]
+
+
+@mock.patch(
+    "snuba.web.views.get_enabled_dataset_names",
+    return_value=["events", "experimental"],
+)
+@mock.patch("snuba.web.views.get_dataset", side_effect=fake_get_dataset)
+def test_check_clickhouse(mock1: mock.MagicMock, mock2: mock.MagicMock) -> None:
+    assert check_clickhouse(filter_experimental=True)
+    assert not check_clickhouse(filter_experimental=False)
```

Comment:
You could also add an assertion that the healthcheck does fail when the dataset is really misconfigured

Reply:
Wouldn't this already do that?
```
assert not check_clickhouse(filter_experimental=False)
```

the dataset is misconfigured and because we have not ignored it the healthcheck fails

Reply:
Or you want a non-experimental dataset to be misconfigured and for it to fail as well?

Reply:
Yeah, basically the test against the function's original behavior (which we never had before)

Reply:
Sure let's be diligent

Reply:
done